var/spack/repos/builtin/packages/py-melissa-core/package.py

```diff
@@ -31,17 +33,22 @@ class PyMelissaCore(PythonPackage):
     depends_on("py-setuptools@46.4:", type=("build"))
     # requirements.txt (SA)
     depends_on("py-pyzmq@22.3.0:", type="run")
-    depends_on("py-mpi4py@3.1.3:", type="run")
+    depends_on("py-mpi4py@3.1.3", type="run")
     depends_on("py-numpy@1.21:", type="run")
     depends_on("py-jsonschema@4.5:", type="run")
     depends_on("py-python-rapidjson@1.8:", type="run")
     depends_on("py-scipy@1.10.0:", type="run")
-    depends_on("py-cloudpickle@2.2.0:", type="run")
+    depends_on("py-plotext@5.2.8:", type="run")
+    depends_on("py-cloudpickle@2.2.0:", type="run", when="@joss:")
```

Comment:
The trailing colon isn't reliable for named tags since they're simply sorted lexicographically. In this case, you want something like:
```suggestion
    depends_on("py-cloudpickle@2.2.0:", type="run", when="@develop,joss")
```